pilot/pkg/model/network.go

```diff
@@ -44,6 +45,10 @@ type NetworkGateway struct {
 	Addr string
 	// gateway port
 	Port uint32
+	// HBONEPort if non-zero indicates that the gateway supports HBONE
+	HBONEPort uint32
```

Comment:
This is not a general used attribute for a gateway, i think for ambient only. So can we use port only, and make 15008 a magic port.

For ambient mode, i donot think it can forward to a gateway that doenot support HBONE.

Reply:
We could maybe re-use `Port` but then we would need some way to indicate it is HBONE supported, like another bool?

Reply:
In ambient mode, can we support non HBONE network gateway